src/main/java/org/apache/sling/jcr/repoinit/impl/NodePropertiesVisitor.java

```diff
@@ -118,23 +116,25 @@ public NodePropertiesVisitor(Session s) {
      * is the same as the autocreated default value
      *
      * @param n the node to check
-     * @param pRelPath the property relative path to check
+     * @param propertyPath the property relative path to check
      * @return true or false
      */
-    protected static boolean isUnchangedAutocreatedProperty(Node n, final String pRelPath)
+    protected static boolean isUnchangedAutocreatedProperty(Node n, final String propertyPath)
             throws RepositoryException {
         boolean sameAsDefault = false;
 
         // deal with the pRelPath nesting
-        Path path = Paths.get(pRelPath);
-        Path parentPath = path.getParent();
-        String name = path.getFileName().toString();
-        if (parentPath != null) {
-            String relPath = parentPath.toString();
-            if (n.hasNode(relPath)) {
-                n = n.getNode(relPath);
-            } else {
-                n = null;
+        final int pos = propertyPath.lastIndexOf("/");
```

Comment:
just checking: we know that there's no trailing "/" here?

Reply:
addressed by moving to `org.apache.jackrabbit.util.Text`

Reply:
... and then finally addressed by using the JCR API instead of reimplementing its functionality.